### ISSUE TITLE
5 panels 32x16

### ISSUE THREAD

**Issue by ostium** [hzeller/rpi-rgb-led-matrix#552]
Hello,
I'm trying to build a video-wall with 5 panels 16x32, like image below, with 480mm (length) x 192mm (height).

![whatsapp image 2018-03-08 at 17 04 13](https://user-images.githubusercontent.com/29130312/37164650-d0f42156-22f2-11e8-8195-0216e43a9a25.jpeg)

It is possible to do this? 

I have all panels wired up, and i can execute "sudo examples-api-use/demo -D0" without problems, but now I don't know how to configure this setup.

Please, can someone help me?

Thanks in advance :-)

**Comment by hzeller** [comment]
Yes, you can do this. I suppose you have 5 panels with 32x16 and you chain them all together in one chain ? Your parameters would be
```
--led-rows=16 --led-cols=32 --led-chain=5
```
Depending on if these are outdoor panels or not, you might need to have to choose a`--led-multiplexing=xxx`parameter.

Given that you have a the LED panels not in their natural orientation (which is over the 32 width, and clocked from a Pi on the right), you need to [write a PixelMapper](https://github.com/hzeller/rpi-rgb-led-matrix/tree/master/examples-api-use#remapping-coordinates) that does the coordinate mapping.

**Comment by ostium** [comment]
My setup is this: 80 columns and 32 lines:
![whatsapp image 2018-03-08 at 17 28 03](https://user-images.githubusercontent.com/29130312/37166200-560cc2a0-22f6-11e8-965b-b81d9b7e703b.jpeg)

but, if I execute this, the image is not in landscap format:
sudo ./demo -t 10 -D 1 runtext.ppm --led-gpio-mapping=classic-pi1 --led-rows=16 --led-cols=32 --led-chain=5

![whatsapp image 2018-03-08 at 17 31 34](https://user-images.githubusercontent.com/29130312/37166346-b5e40fee-22f6-11e8-9387-024602640e79.jpeg)

Maybe is a rotational problem on software? Where I can change this?

Thanks

**Comment by ostium** [comment]
Sorry, I didn't read all off your response... "you need to write a PixelMapper that does the coordinate mapping."

**Comment by angelogoncalve** [comment]
Use this transformer to 32x16 panel:

void SArrangementTransformer::TransformCanvas::SetPixel(
  int x, int y, uint8_t red, uint8_t green, uint8_t blue) {
  if (x < 0 || x >= width_ || y < 0 || y >= height_) return;
  const int slab_height = virtualrows_*panel_height_;   // one folded s-shape
  const int base_y = (y / slab_height) * panel_height_;
  y %= slab_height;
   if (y < panel_height_) {
   x += delegatee_->width()*2/3 ;

     }
 if ((y < 2*panel_height_ ) && (y >= panel_height_)) {
    x = 2*width_ - x - 1;
    y = slab_height *2/3 - y - 1;

  }

 if  ((y < 3*panel_height_ ) && (y >=2* panel_height_)) {
    
     y =  y - slab_height*2/3 ;

  }
  delegatee_->SetPixel(x, base_y + y, red, green, blue);
}

**Comment by angelogoncalve** [comment]
You can try to rotate the image doing this function:
    
def rotate_image(self, image, angle):
return image.convert("RGB").rotate(angle)

**Comment by ostium** [comment]
@angelogoncalve 

Hello, thank you for the help... your #611 save my day! It works very well!

**Comment by marcmerlin** [comment]
With apologies to everyone, this project is doing a one time cleanup of 
all the bugs that have not been addressed and closed in the last 10 years.

If you filed an issue to ask for help, please do not re-open this one or a new
bug, but go here instead: https://rpi-rgb-led-matrix.discourse.group/

Please be mindful that this project does not have any full time maintainers
or anyone paid. You should not expect personal help for your own projects or
answers to programming questions but you can ask for help with panels that
aren't working, setup questions, or anything that may be a bug here:
https://rpi-rgb-led-matrix.discourse.group/
If you need programming help and are willing to pay someone for it, please
state so in your message, and someone may be interested.

If you did report an actual bug that is still not fixed today, please accept
our apologies for this being mass closed, you can absolutely send a followup
and re-open the issue, or open a new up to date one as applicable.
Note that we very much welcome help, so if you can help fix your bug, you will
be a superstar :)

If your bug was linked to a pull request that was not resolved, please update
the pull request and the issue, and we will do our best to review them. We do
thank you for your undertanding that we have very limited free time, so any
extra work you do along with your bug report or PR request will be very much
appreciated and improve the odds that they will be looked at.

If you want to help contribute as a python or C# or any other language bindings,
you are also a superstar, please state so in the issue and we'll do our best to
give you the access and reviews you need.

For anything related to PWM panel support, please make sure you read README and
then refer to https://github.com/hzeller/rpi-rgb-led-matrix/issues/466 . 
Please do not comment if you're haven't read everything and/or if you're just
asking when your panel is going to be supported. Those bugs are if you're 
able to contribute to the effort.

For rPI5: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1603

For anything python, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
For anything C#, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
